src/emulator/np/src/trophy/context.cpp

```diff
@@ -0,0 +1,373 @@
+                    if (trophy_progress[i] & (1 << j)) {
+                        total++;
+                    }
+                }
+            }
+        }
+    }
+
+    return total;
+}
+
+bool Context::get_trophy_description(const std::int32_t id, std::string &name, std::string &detail) {
+    if (id < 0 || id >= MAX_TROPHIES) {
+        return false;
+    }
+
+    if (trophy_detail_xml.empty()) {
+        const std::string fname = fmt::format("TROP_{:0>2d}.SFM", lang);
+
+        if (!read_trophy_entry_to_buffer(trophy_file, fname.c_str(), trophy_detail_xml)) {
+            if (!read_trophy_entry_to_buffer(trophy_file, "TROP.SFM", trophy_detail_xml)) {
+                return false;
+            }
+        }
+    }
+
+    // Parse it
+    pugi::xml_document doc;
+    const auto result = doc.load_string(trophy_detail_xml.c_str());
+
+    if (!result) {
+        return false;
+    }
+
+    // Try to find the description for the id
+    for (auto trop: doc.child("trophyconf")) {
+        if ((strncmp(trop.name(), "trophy", 6) == 0) && (trop.attribute("id").as_uint() == id)) {
+            name = trop.child("name").text().as_string();
+            detail = trop.child("detail").text().as_string();
+
+            return true;
+        }
+    }
+
+    return false;
+}
+}
+
+emu::np::trophy::ContextHandle create_trophy_context(NpState &np, IOState &io, const std::string &pref_path, 
+    const emu::np::CommunicationID *custom_comm, const std::uint32_t lang, emu::np::NpTrophyError *error) {
+    if (!custom_comm) {
+        custom_comm = &np.comm_id;
+    }
+
+    if (error)
+        *error = emu::np::NpTrophyError::TROPHY_ERROR_NONE;
+
+    #define TROPHY_RET_ERROR(err)                                                            \
+        if (error) *error = emu::np::NpTrophyError::err;                                     \
+        return emu::np::trophy::INVALID_CONTEXT_HANDLE
+
+    // Check if a context has already been created for this communication ID 
+    for (std::size_t i = 0; i < np.trophy_state.contexts.size(); i++) {
+        if (np.trophy_state.contexts[i].valid && np.trophy_state.contexts[i].comm_id == *custom_comm) {
+            TROPHY_RET_ERROR(TROPHY_CONTEXT_EXIST);
+        }
+    }
+
+    const std::string trophy_base_dir = fmt::format("app0:sce_sys/trophy/{}_{:0>2d}/", custom_comm->data,
+        custom_comm->num);
+
+    // Initialize the stream
+    const std::string trophy_file_path =  trophy_base_dir + "TROPHY.TRP";
+
+    // Try to open the file
+    const SceUID trophy_file = open_file(io, trophy_file_path, SCE_O_RDONLY, pref_path.c_str(), "create_trophy_context");
+
+    if (trophy_file < 0) {
+        TROPHY_RET_ERROR(TROPHY_CONTEXT_FILE_NON_EXIST);
+    }
+
+    // Try to open the trophy save file. The context will automatically took the default profile to perform trophy
+    // operations on.
+    std::string trophy_progress_save_file = fmt::format("ux0:/user/{}/{}_{:0>2d}/", 
```

Comment:
```suggestion
    std::string trophy_progress_save_file = fmt::format("ux0:/user/00/trophy/{}_{:0>2d}/", custom_comm->data, custom_comm->num); 
```

Reply:
Is that where trophy are store on real vita

Reply:
I made this for multi-user

Reply:
Multi-user has different trophy progress file. If you like me to store in ux0:/user/00/trophy/ than yeah but im not gonna drop the user online ID that easy :p

Reply:
No, i have explain and see on real hardware, PSVITA using only ux0:/user/00/trophy/etc.. for and same inside "ur0:/" for config file trophy.
and if multi user exist, it using 00 and after 01 like ps3

Reply:
Oki

Reply:
I will reflect your said

Reply:
ah yes, for 00 is just for moment :D i works also on multi user suport, have good progress now for moment :)

Reply:
I already have multi-user support